src/Compilers/CSharp/Portable/Symbols/Source/ExtensionGroupingInfo.cs

```diff
@@ -158,6 +160,53 @@ public Cci.ITypeDefinition GetCorrespondingGroupingType(SourceNamedTypeSymbol ty
             throw ExceptionUtilities.Unreachable();
         }
 
+        internal ImmutableArray<SourceNamedTypeSymbol> GetMatchingExtensions(SourceNamedTypeSymbol extension)
+        {
+            Debug.Assert(extension.IsExtension);
+            return GetCorrespondingMarkerType(extension).UnderlyingExtensions;
+        }
+
+        // Returns all the extension blocks but grouped/merged by equivalency (ie. same marker name)
+        internal IEnumerable<ArrayBuilder<SourceNamedTypeSymbol>> EnumerateMergedExtensionBlocks()
+        {
+            Dictionary<string, MultiDictionary<string, SourceNamedTypeSymbol>> groupingMap = this._groupingMap;
+            ArrayBuilder<string> groupingNames = sortStrings(groupingMap.Keys);
+
+            foreach (var groupName in groupingNames)
+            {
+                MultiDictionary<string, SourceNamedTypeSymbol> markerMap = groupingMap[groupName];
+                ArrayBuilder<string> markerNames = sortStrings(markerMap.Keys);
+
+                foreach (var markerName in markerNames)
+                {
+                    ArrayBuilder<SourceNamedTypeSymbol> extensions = sortSourceNamedTypeSymbols(markerMap[markerName]);
```

Comment:
>sortSourceNamedTypeSymbols(markerMap[markerName]); [](http://example.com/codeflow?start=69&length=50)

Note that ```GetMatchingExtensions``` effectively gets the same set as an immutable array, which is already sorted in lexical declaration order.

Reply:
Thanks. Simplified